lms/db/__init__.py

```diff
@@ -51,7 +46,7 @@ def update_from_dict(self, data, skip_keys=None):
                 f"Expected a set of keys to skip but found '{type(skip_keys)}'"
             )
 
-        columns = {col.key for col in self.iter_columns()}
+        columns = set(self.columns())
```

Comment:
This would have been a good change before as well:

` set(self.iter_colums())`